src/Controls/src/Core/Handlers/Items/iOS/CarouselViewController.cs

```diff
@@ -17,20 +18,34 @@ public class CarouselViewController : ItemsViewController<CarouselView>
 		protected readonly CarouselView Carousel;
 
 		CarouselViewLoopManager _carouselViewLoopManager;
-		bool _initialPositionSet;
-		bool _updatingScrollOffset;
+		bool _isCenteringItem;
 		List<View> _oldViews;
 		int _gotoPosition = -1;
 		CGSize _size;
 		ILoopItemsViewSource LoopItemsSource => ItemsSource as ILoopItemsViewSource;
 		bool _isDragging;
+		DisplayOrientation _initialOrientation;
+
+		bool _isRotating;
 
 		public CarouselViewController(CarouselView itemsView, ItemsViewLayout layout) : base(itemsView, layout)
 		{
 			CollectionView.AllowsSelection = false;
 			CollectionView.AllowsMultipleSelection = false;
 			itemsView.Scrolled += CarouselViewScrolled;
 			_oldViews = new List<View>();
+			_initialOrientation = DeviceDisplay.MainDisplayInfo.Orientation;
+			DeviceDisplay.MainDisplayInfoChanged += OnDisplayInfoChanged;
```

Comment:
Can we move this subscription to a lifecycle event? Maybe `WillMoveToParentViewController`

so it's only subscribed to when attached to the visual tree?

Also, I think you can use `ViewWillTransitionToSize` to detect if something is rotating and wire into the coordinator

```C#
coordinator.AnimateAlongsideTransition((obj) => 
    {
        // This block will be executed during the rotation animation.
        // You can add your own code here to respond to the rotation.
    }, (obj) => 
    {
        // This block will be executed after the rotation animation completes.
        // You can add your own code here to respond to the completion of the rotation.
    });
```